apis/duck/v1alpha1/util.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"encoding/json"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// FromUnstructured tkaes unstructured object from (say from client-go/dynamic) and
+// converts it into our duck types.
+func FromUnstructured(obj unstructured.Unstructured, target interface{}) error {
```

Comment:
Oops, I meant to push this up a layer in the directory structure, since I don't see a strong reason to version this.

Reply:
I don't think I can just move it there, because I need the types from v1alpha1 directory which imports ../ and then the test would fail. I moved it to apis/duck/util